lib/galaxy/tools/deps/resolvers/__init__.py

```diff
@@ -1,7 +1,17 @@
-from abc import ABCMeta, abstractmethod
+from galaxy.util.dictifiable import Dictifiable
 
+from ..requirements import ToolRequirement
 
-class DependencyResolver( object ):
+from abc import ABCMeta, abstractmethod, abstractproperty
+
+
+class DependencyResolver(Dictifiable, object):
+    # Keys for dictification.
+    dict_collection_visible_keys = ['resolver_type', 'resolves_simple_dependencies']
+    # A "simple" dependency is one that does not depend on the the tool
+    # resolving the dependency. Classic tool shed dependencies are non-simple
+    # because the tool shed install.
```

Comment:
`because the tool shed install`? Because the tool shed needs to resolve the dependencies?